### PR TITLE
Return errors as JSON if accepted but HTML not

### PR DESCRIPTION
**Pull request by Martchus**
Fix https://progress.opensuse.org/issues/14898

### CODE REVIEW

File: lib/OpenQA/WebAPI.pm [hunk 1]
```diff
@@ -473,6 +473,17 @@ sub startup {
...
     }
+
+    # add method to be called before rendering
+    $self->app->hook(
+        before_render => sub {
+            my ($c, $args) = @_;
+            # return errors as JSON if accepted but HTML not
+            if (!$c->accepts('html') && $c->accepts('json') && $args->{status} && $args->{status} != 200) {
+                # the JSON API might already provide JSON in some error cases which must be preserved
+                (($args->{json} //= {})->{error_status}) = $args->{status};
```
Comment (Martchus) [inline lib/OpenQA/WebAPI.pm]: I can split this into multiple lines, but it is actually quite compact this way.

Comment (kraih) [inline lib/OpenQA/WebAPI.pm]: The outer parentheses should not be necessary.

**Comment by Martchus** [comment]
I included `Fix fedmsg test` from other PR so Travis won't fail anymore.

**Comment by coolo** [comment]
merged locally

**Pull request closed without merging**